### PR TITLE
Cloud code logs

### LINKED ISSUE #2244: Cloud Code logs not working; VERBOSE=1 does work
I have setup parse server on AWS. I have connected my iOS app to it however I am getting errors returned to the xcode console. This code did not error on parse.com hosted server. In order for me to track down what the error is I need to see the cloud code logs.

I have set up parse dashboard locally and it connects however, there are no logs. Before you close this as a duplicate: I have setup FileAdapterLogger as stated in --> https://github.com/ParsePlatform/parse-server/issues/781

My elastic beanstalk server is healthy and the app calls the CC. If I use "verbose: 1" in my index.js I get a TON of output into the log files in parse dashboard. I need to see just the cloud code logs including errors in the error log. 

This is my setup:
```
var logger = require('parse-server/lib/Adapters/Logger/FileLoggerAdapter').FileLoggerAdapter;
  
var api = new ParseServer({
  databaseURI: databaseUri || 'mongodb://xxxxxxxxxxxxxxxxxxxxxxxx',
  cloud: process.env.CLOUD_CODE_MAIN || __dirname + '/cloud/main.js',
  appId: process.env.APP_ID || 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
  masterKey: process.env.MASTER_KEY || 'xxxxxxxxxxxxxxxxxxxxxxxxxx',
  serverURL: process.env.SERVER_URL || 'http://xxxxxxxxxxxxxxxxxxxx/parse/',
  loggerAdapter: new logger({ logsFolder: './' })
});
```

#### Expected Results

I need to have cloud code logs working in locally hosted dashboard.

#### Actual Outcome

"No logs in the last 30 days"

### Environment Setup

- **Server**
  - parse-server version: latest
  - Operating System:    latest
  - Hardware:             latest
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): AWS

- **Database**
  - MongoDB version: latest
  - Storage engine: latest
  - Hardware:        latest
  - Localhost or remote server? (AWS, mLab, ObjectRocket, Digital Ocean, etc): mLab

I've set this all up this week so every version is the latest

### PR DESCRIPTION
**Pull request by flovilmart**
Fixes #2244

- Adds info log when running triggers and functions
- Adds error logs when trigger or function fails
- Match original parse.com log implementation

### CODE REVIEW

File: src/rest.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ function del(config, auth, className, objectId, clientSDK) {
 // Returns a promise for a {response, status, location} object.
 function create(config, auth, className, restObject, clientSDK) {
   enforceRoleSecurity('create', className, auth);
-  var write = new RestWrite(config, auth, className, null, restObject, clientSDK);
+  var write = new RestWrite(config, auth, className, null, restObject, null, clientSDK);
```
Comment (TylerBrock) [inline src/rest.js]: nice catch

Comment (flovilmart) [inline src/rest.js]: I started having SDK and version set on those objects :/

**Comment by tanmays** [comment]
@flovilmart just tried the latest 2.2.17 release, how do I configure it to show just the messages from `console.log` statements? Right now its showing the entire request-response even with verbose off. Even if I explicitly set verbose to false it still logs in verbose.

**Comment by TylerBrock** [comment]
@tanmays thanks for reporting this, I noticed the same thing and unfortunately you can't turn them off currently. This is so the cloud code logs work in the dashboard, it's only showing the requests and responses of the cloud code functions and not every request to parse-server.

In the future hopefully we will try to work the code so that the logs for the cloud code needed for the dashboard aren't also logged to the console.

**Comment by tanmays** [comment]
@TylerBrock thanks! Hopefully we can achieve the same behaviour as Parse.com dash where only console statements are logged and not req/res (unless verbose is on). Right now 99% of my code is via cloud code and my console log is barely readable. :(

**Comment by Siedrix** [comment]
@tanmays Im having the same problem, code changes on FunctionsRouter.js now make the console unreadable, both on dev and on production. 

I would suggest a way to set winston log level to warn for now:
```
var winston = require('winston');
winston.level = 'warn';
```
If you add this code before the `new ParseServer({conf}),` you prevent all this logging.

**Comment by joeyslack** [comment]
@Siedrix You saved my life! I was hacking around in the logging adapters, and found this fix to be much much better.

Since 2.2.18 release is completely broken, being able to fix the logging on 2.2.17 has saved me an enormous amount of money in logging costs (hosted logging).

**Comment by flovilmart** [comment]
You should be able to set the logLevel on Parse server initialisation with 2.2.18.

When you say completely broken, can you open more issues so we can cook 2.2.19 ASAP

**Comment by joeyslack** [comment]
Hey there,

Along with: 
https://github.com/ParsePlatform/parse-server/issues/2570
, header issues, dashboard integration issues, https and piggyback ssl issues, “parse-server” cannot be found, I’m not even sure if we have anyone testing 2.2.18? On a fresh install, I simply cannot get it to work in a dev environment. Perhaps I’m doing something wrong, but I guess I don’t see what everyone else does. parse-server just refuses to be loaded, cannot be found in path.

On Wed, Aug 24, 2016 at 12:58 PM Florent Vilmart

<
mailto:Florent Vilmart <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

You should be able to set the logLevel on Parse server initialisation with 2.2.18.

When you say completely broken, can you open more issues so we can cook 2.2.19 ASAP

—

You are receiving this because you commented.

Reply to this email directly,
https://github.com/ParsePlatform/parse-server/pull/2370#issuecomment-242188651
, or

**Comment by flovilmart** [comment]
@joeyslack the more contributions, the faster it will be solved :)

**Comment by sprabs** [comment]
Sorry if this is the wrong place to post... 

I'm guessing this has to be something with my setup, but I can no longer see logs other than those from the parse-server-push-adapter, new objects created, and console.log statements explicitly in the Cloud Code. I really want to see errors and info, but even verbose on, it's not working:

verb parse-server-push-adapter GCM             "message_id": "0:1472668642692974%a8fa1296f9fd7ecd"
verb parse-server-push-adapter GCM         },
verb parse-server-push-adapter GCM         {
verb parse-server-push-adapter GCM             "message_id": "0:1472668642694754%a8fa1296f9fd7ecd"
verb parse-server-push-adapter GCM         },
verb parse-server-push-adapter GCM         {
verb parse-server-push-adapter GCM             "message_id": "0:1472668642692145%a8fa1296f9fd7ecd"
verb parse-server-push-adapter GCM         }
verb parse-server-push-adapter GCM     ]
verb parse-server-push-adapter GCM }
ParseObjectSubclass { className: 'Contacts', _objCount: 1162, id: '0WFUr5CWQV' }


It was fine until I did a pull on master in the last week. I noticed there have been quite a few PRs for logging recently. With these changes are there any additional things we need to add to see the logs? I currently have this line in index.js:

var logger = require('parse-server/lib/Adapters/Logger/FileLoggerAdapter').FileLoggerAdapter;

and inside the new ParsServer block: 
loggerAdapter: new logger({ logsFolder: './' })

I'm running my app on AWS Elastic Beanstalk.

**Pull request merged**